[Serializer] Add an @Ignore annotation #28744

vasilvestre opened the pull request:
Complete https://github.com/symfony/symfony/pull/28744
Fix https://github.com/symfony/symfony-docs/issues/13585

javiereguiluz commented on components/serializer.rst:
```diff
@@ -413,6 +413,80 @@ As for groups, attributes can be selected during both the serialization and dese
 Ignoring Attributes
 -------------------
 
+Set `false` by default, you can ignore attribute by two methods.
```
Code inside paragraphs must use double backticks:

```rst
Set ``false`` by default ...
```

javiereguiluz replied:
I don't understand this -> "Set `false` by default, ..."

Maybe we can reword it like this:

```rst
All attributes are included by default when serializing objects. You have
two alternatives to ignore some of those attributes.
```

javiereguiluz commented on components/serializer.rst:
```diff
@@ -413,6 +413,80 @@ As for groups, attributes can be selected during both the serialization and dese
+* `Option 1: Using annotation @Ignore`_
+* `Option 2: Using the context`_
+
+Option 1: Using annotation @Ignore
```
```rst
Option 1: Using ``@Ignore`` annotation
--------------------------------------
```

javiereguiluz commented on components/serializer.rst:
```diff
@@ -413,6 +413,80 @@ As for groups, attributes can be selected during both the serialization and dese
+
+
+Option 2: Using the context
+---------------------------------
```
The underline should be as long as the headline text.

javiereguiluz commented on components/serializer.rst:
```diff
@@ -413,6 +413,80 @@ As for groups, attributes can be selected during both the serialization and dese
+Option 2: Using the context
+---------------------------------
+
 As an option, there's a way to ignore attributes from the origin object.
```
This phrase needs some update:

```
As an option, there's a way to ignore attributes from the origin object.
```

Because now this is not the only way of achieving this and we already explained another way of doing it before.

vasilvestre replied:
Wdyt about new sentence ?

javiereguiluz reviewed (commented):
@vasilvestre thanks for contributing these docs. I left some comments about some minor things, but overall it looks OK to me 👍

jennevdmeer commented on components/serializer.rst:
```diff
@@ -413,8 +413,81 @@ As for groups, attributes can be selected during both the serialization and dese
+
+        Acme\MyObj:
+            attributes:
+                foo:
+                    groups:
+                        ignore: false
+                bar:
+                    groups:
+                        ignore: true
```
Ignore should probably be listed on annotation level (as a direct child of the attribute name) instead of as being a group.

noniagriconomie commented on components/serializer.rst:
```diff
@@ -413,8 +413,79 @@ As for groups, attributes can be selected during both the serialization and dese
+
+    .. code-block:: php-annotations
+
+        namespace Acme;
```
```suggestion
        namespace App/Model;
```

noniagriconomie commented on components/serializer.rst:
```diff
@@ -413,8 +413,79 @@ As for groups, attributes can be selected during both the serialization and dese
+
+        use Symfony\Component\Serializer\Annotation\Ignore;
+
+        class MyObj
```
```suggestion
        class MyClass
```

and rename accordingly other code below

noniagriconomie commented on components/serializer.rst:
```diff
@@ -413,8 +413,79 @@ As for groups, attributes can be selected during both the serialization and dese
+        class MyObj
+        {
+            public $foo;
+            /**
```
blank line before :)

OskarStark replied:
```suggestion

            /**
```

OskarStark commented on components/serializer.rst:
```diff
@@ -413,8 +413,79 @@ As for groups, attributes can be selected during both the serialization and dese
+            </class>
+        </serializer>
+
+You are now able to ignore specifics attributes during serialization::
```
```suggestion
You are now able to ignore specific attributes during serialization::
```

noniagriconomie commented on components/serializer.rst:
```diff
@@ -413,8 +413,80 @@ As for groups, attributes can be selected during both the serialization and dese
+
+You are now able to ignore specific attributes during serialization::
+
+    use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
```
`use App\Model\MyClass`

noniagriconomie commented on components/serializer.rst:
```diff
@@ -413,8 +413,80 @@ As for groups, attributes can be selected during both the serialization and dese
+    use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+    use Symfony\Component\Serializer\Serializer;
+
+    $obj = new MyObj();
```
```suggestion
    $obj = new MyClass();
```

noniagriconomie replied:
and inside the config values as well :)

noniagriconomie reviewed (commented):
small code changes required after renaming `MyObj` in favor of `MyClass`

Commit:
[Serializer] Add an @Ignore annotation #28744


Update serializer.rst
Update components/serializer.rst

Co-authored-by: Oskar Stark <oskarstark@googlemail.com>
Update components/serializer.rst

Co-authored-by: Antoine Makdessi <antoine.makdessi@agriconomie.com>
Update components/serializer.rst

Co-authored-by: Antoine Makdessi <antoine.makdessi@agriconomie.com>
Update components/serializer.rst

Co-authored-by: Oskar Stark <oskarstark@googlemail.com>
Update components/serializer.rst

Co-authored-by: Antoine Makdessi <antoine.makdessi@agriconomie.com>
Update serializer.rst

vasilvestre commented:
@javiereguiluz I think this should be good now !

javiereguiluz commented on components/serializer.rst:
```diff
@@ -413,8 +413,81 @@ As for groups, attributes can be selected during both the serialization and dese
+* `Option 2: Using the context`_
+
+Option 1: Using ``@Ignore`` annotation
+--------------------------------------
```
We should change `------` by `~~~~~~` here and in the other headline to make them both sub-headlines of the main `Ignoring Attributes` headline.

vasilvestre committed:
Update serializer.rst

vasilvestre committed:
Update serializer.rst

Remove default value (false) from ignoring rules in XML and YML files as I didn't put them in PHP code.

javiereguiluz commented:
Valentin, thanks a lot for your contribution ... and congrats on your very first contribution to Symfony Docs 🎉
